### PR TITLE
DashboardScenes: Default to first variable value when editing a repeated panel

### LINKED ISSUE #91188: DashboardScenes: Editing the original panel in repeated panels breaks the query
In a repeated panel using a multi variable, when trying to edit the original panel the query created will contain a string of all selected variable values, instead of using one single value for that particular panel. This may cause the query to be invalid and fail, thus showing `No Data`, while the panels do show correct data while in the dashboard, which causes confusion between dashboard and panel edit.

### PR DESCRIPTION
**Pull request by mdvictor**
When using repeated panels, the queries are build properly in the dashboard, each repeated panel using the correct variable value in the query. But when editing the original panel the query is built using all variable values, causing it to return `No Data`. This PR fixes that by cloning the variable at the `VizPanelManager` level and defaulting to the first value so the query runs and shows the correct data for the original panel.

Closes https://github.com/grafana/grafana/issues/91188

### CODE REVIEW

File: public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx [hunk 1]
```diff
@@ -97,6 +101,27 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
...
+          name: variable.state.name,
+          value: values[0],
+          text: String(texts[0]),
+        });
+        if (!variables) {
+          variables = new SceneVariableSet({
+            variables: [varWithDefaultValue],
+          });
+        } else {
+          variables.state.variables.push(varWithDefaultValue);
```
Comment (ivanortegaalba) [inline public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx:120]: Should we do `variables.setState({variables})` here? It won't trigger re-render in components subscribed to this state. Is this on purpose?

Comment (mdvictor) [reply public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx:120]: Yeah, it should use setState, thank you!

**Review by ivanortegaalba — APPROVED** [review]
Ey Victor! This works like a charm! Great work on this! 💯

**Comment by ivanortegaalba** [comment]
I added the backport label, to get it fixed in 11.2.0 too 😄

**Pull request merged**